pkg/skaffold/deploy/kpt.go

```diff
@@ -111,3 +124,35 @@ func kptCommandArgs(dir string, commands, flags, globalFlags []string) []string
 
 	return args
 }
+
+// getResources returns a list of all file names in `root` that end in .yaml or .yml
+// and all local kustomization dependencies under root.
+func getResources(root string) ([]string, error) {
+	var files []string
+
+	if _, err := os.Stat(root); os.IsNotExist(err) {
+		return nil, err
+	}
+
+	err := filepath.Walk(root, func(path string, info os.FileInfo, _ error) error {
+		isResource, err := regexp.MatchString(`\.ya?ml$`, filepath.Base(path))
```

Comment:
This is fair enough for now. But it is actually not accurate because the content of the .yaml files should also be KRM based. Can you add a comment saying that we expect kpt to provide a more accurate functions to walk through a dir and pulls out the kustomize/kpt resources.

Reply:
Sure thing. I agree, this may be something that can be offered when we move away from consuming kpt as a CLI.